### PR TITLE
9th step

### PR DESCRIPTION
**Pull request by alemannosara**

### CODE REVIEW

File: src/main/scala/Macro.scala [hunk 1]
```diff
@@ -40,41 +38,127 @@ object FIntepolator extends MacroStringInterpolator[String] {
...
+      * that they refer to 
+      * For example : f2"${d}random-leading-junk%d" will lead to an error
       */
-    def addDefaultFormat(partsExpr : List[Expr[String]]) : List[String] = {
-       val parts : List[String] = partsExpr.map{case Literal(str : String) => str} //TODO : do it once, use the pair //TODO : case _ => return None in the other function
-       parts match {
+    def addDefaultFormat(partsExpr : List[Expr[String]]) : Option[List[String]] = {
+      val parts : List[String] = partsExpr.map{
+        case Literal(str : String) => str
+        case _ => return None
```
Comment (alemannosara) [inline src/main/scala/Macro.scala]: @nicolasstucki
Some tests are not passing. I am wondering whether this is due to the pattern matching here. Indeed, for the negative test numberArgumentsTest2, the "case Literal" is never entered. Do you think I should change "Literal" with "Const"?
I have tried to do so but it seems not to recognize it. Any idea?

Comment (nicolasstucki) [reply src/main/scala/Macro.scala]: You could try `case scala.quote.matcher.Literal(...`

Comment (nicolasstucki) [reply src/main/scala/Macro.scala]: Or update to the latest nightly

Comment (alemannosara) [reply src/main/scala/Macro.scala]: Thank you

**Comment by alemannosara** [comment]
@nicolasstucki 
I found a new bug and tried to fix it on commit bdf1c2b : "Find new bug : "%% %.2d" and fix it". I changed the way I go through the part if there are more than one referenced argument. However, I cannot compile my files anymore : sbt times out. 
I have really no clue on why it does so, any idea?

**Comment by nicolasstucki** [comment]
Maybe try `sbt clean` or delete the `target` folder

**Comment by nicolasstucki** [comment]
Or maybe you had an infinite loop in the macro implementation

**Pull request merged**